src/zenml/orchestrators/step_run_utils.py

```diff
@@ -554,13 +554,14 @@ def link_output_artifacts_to_model_version(
         output_configurations: The output configurations for the step.
         model_version: The model version to link.
     """
-    for output_name, output_artifact in artifacts.items():
-        artifact_config = None
-        if output_config := output_configurations.get(output_name, None):
-            artifact_config = output_config.artifact_config
-
-        link_artifact_version_to_model_version(
-            artifact_version=output_artifact,
-            model_version=model_version,
-            artifact_config=artifact_config,
-        )
+    for output_name, output_artifacts in artifacts.items():
+        for output_artifact in output_artifacts:
+            artifact_config = None
+            if output_config := output_configurations.get(output_name, None):
```

Comment:
We need to make sure that this function is only being called with regular step output artifacts, as otherwise the artifact config does not apply

Reply:
This is currently not the case

Reply:
I fixed this, but one thing we don't (and never did): If a manual save was a model artifact for the original step run, it will not be a model artifact in the cached step run.

Reply:
I pushed a test for this case, which is skipped now. We will handle this separately, as discussed.